kolibri/plugins/learn/assets/src/views/TopicsPage/FloatingSidePanel.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+
+  <FullScreenSidePanel
+    class="full-screen-side-panel"
+    alignment="left"
+    :fullScreenSidePanelCloseButton="true"
+    sidePanelOverrideWidth="300px"
+    @closePanel="$emit('close')"
+    @shouldFocusFirstEl="findFirstEl()"
+  >
+    <KIconButton
+      v-if="windowIsSmall && currentCategory"
+      icon="back"
+      :ariaLabel="coreString('back')"
+      :color="$themeTokens.text"
+      :tooltip="coreString('back')"
+      @click="$emit('closeCategoryModal')"
+    />
+    <EmbeddedSidePanel
+      v-if="!currentCategory"
+      ref="embeddedPanel"
+      topicPage="True"
+      width="100%"
+      :value="value"
+      :topicsListDisplayed="!mobileSearchActive"
+      :topics="topics"
+      :topicsLoading="topicsLoading"
+      :more="topicMore"
+      :genContentLink="genContentLink"
+      :availableLabels="availableLabels"
+      :activeActivityButtons="activeActivityButtons"
+      :activeCategories="activeCategories"
+      :showChannels="false"
+      position="overlay"
+      @input="val => $emit('searchTerms', val)"
+      @currentCategory="$emit('currentCategory', $event)"
+      @loadMoreTopics="$emit('loadMoreTopics')"
+    />
+  </FullScreenSidePanel>
+
+</template>
+
+
+<script>
+
+  import FullScreenSidePanel from 'kolibri.coreVue.components.FullScreenSidePanel';
+  import genContentLink from '../../utils/genContentLink';
+  import EmbeddedSidePanel from '../EmbeddedSidePanel';
+
+  export default {
+    name: 'FloatingSidePanel',
+    components: { EmbeddedSidePanel, FullScreenSidePanel },
```

Comment:
Also, the names are bloody confusing. 'FullScreenSidePanel' feels like it is in some way taking up the whole screen, when it isn't - embedded side panel is embedded inside the full screen side panel, and now we are nesting both inside a floating side panel?

![image](https://user-images.githubusercontent.com/1680573/165984528-1a9a0ebb-fb34-40f6-bb25-8d26aeb34754.png)

Reply:
I am the one responsible for the fullscreen vs embedded situation... 🙃  I'd thought previously about renaming fullscreen as 'overlaid' or something similar and then never took any action because I wasn't sure how much better it was

Reply:
I'm definitely up for a group chat to figure out what to call these things :smile:

Reply:
I could rename them here to something like (open to suggestions)

**FullScreenSidePanel -> SidePanelModal | DrawerModal | PopOutModal **

**EmbeddedSidePanel -> SearchForm | SearchControlPanel**

**FloatingSidePanel -> SearchControlPanelModal | SearchPopOutModal**

Modal seems to imply the "it pops up, steals focus, lets you do a thing, and you can dismiss it"

PopOut and Drawer to me both say "I come out from the side and go away when you're done with me".

The EmbeddedSidePanel is basically the helm for search controls so something around that functionality seems to be a good way forward.

The FloatingSidePanel is basically just wrapping EmbeddedSidePanel with FullScreenSidePanel... so its name might be best decided after the other two are :laughing: